src/openeo_test_suite/tests/processes/processing/test_example.py

```diff
@@ -154,110 +202,117 @@ def prepare_argument(arg, process_id, name, connection, file):
+                    (example[key], result[key]) = _prepare_results(
+                        connection=connection,
+                        file=file,
+                        example=example[key],
+                        result=result[key],
                     )
 
     elif isinstance(example, list):
+        # TODO: avoid in-place list mutation
         for i in range(len(example)):
             if i >= len(result):
-                (example[i], _) = prepare_results(connection, file, example[i])
+                (example[i], _) = _prepare_results(
+                    connection=connection, file=file, example=example[i]
+                )
             else:
-                (example[i], result[i]) = prepare_results(
-                    connection, file, example[i], result[i]
+                (example[i], result[i]) = _prepare_results(
+                    connection=connection,
+                    file=file,
+                    example=example[i],
+                    result=result[i],
                 )
 
     return (example, result)
 
 
-def load_ref(ref, file):
+def _load_ref(ref: str, file: Path):
     try:
-        path = posixpath.join(file.parent, ref)
+        path = file.parent / ref
         if ref.endswith(".json") or ref.endswith(".json5") or ref.endswith(".geojson"):
             with open(path) as f:
                 return json5.load(f)
         elif ref.endswith(".txt") or ref.endswith(".wkt2"):
             with open(path) as f:
                 return f.read()
         else:
-            raise Exception(
+            raise NotImplementedError(
                 "External references to files with the given extension not implemented yet."
             )
     except Exception as e:
-        raise Exception("Failed to load external reference {}: {}".format(ref, e))
+        # TODO: is this try-except actually useful?
+        raise Exception(f"Failed to load external reference {ref}") from e
 
 
 def check_non_json_values(value):
+    # TODO: shouldn't this check be an aspect of Http(ProcessTestRunner)?
     if isinstance(value, float):
         if math.isnan(value):
-            raise Exception("HTTP JSON APIs don't support NaN values")
+            raise ValueError("HTTP JSON APIs don't support NaN values")
         elif math.isinf(value):
-            raise Exception("HTTP JSON APIs don't support infinity values")
+            raise ValueError("HTTP JSON APIs don't support infinity values")
     elif isinstance(value, dict):
-        for key in value:
-            check_non_json_values(value[key])
+        for v in value.values():
+            check_non_json_values(v)
     elif isinstance(value, list):
         for item in value:
             check_non_json_values(item)
 
 
 def check_exception(example, result):
-    assert isinstance(result, Exception), "Excpected an exception, but got {}".format(
-        result
-    )
+    assert isinstance(result, Exception)
```

Comment:
Why remove the message?

Reply:
because pytest by default provides a useful message already, so you don't need all this boilerplate in your tests.

e.g.

```python

def test_tmp():
    result = "lalala"
    assert isinstance(result, Exception)
```
fails with 
```
>       assert isinstance(result, Exception)
E       AssertionError: assert False
E        +  where False = isinstance('lalala', Exception)

src/openeo_test_suite/tests/test_dummy.py:22: AssertionError
```

Note how pytest mentions the value of `result` in the error report

Reply:
I felt like the messages are easier to understand in comparison what pytest provides by default.